### ISSUE TITLE
CA2000 not triggered for undisposed IEnumerator

### ISSUE THREAD

**Issue by runehalfdan** [dotnet/roslyn-analyzers#5633]
This little snippet, doing the opposite of Any(), did not trigger a CA2000
```
public static bool Empty<TSource>(this IEnumerable<TSource> source)
{
    return !source.GetEnumerator().MoveNext();
}
```

(I suspect this is related  to some ownership transfer rules I'm not familiar with, but if someone could point out those rules it would be helpful)
